sys/security/mac_do/mac_do.c

```diff
@@ -2023,6 +2020,16 @@ rule_grant_supplementary_groups(const struct rule *const rule,
 	return (0);
 }
 
+static int
+rule_grant_supplementary_groups(const struct rule *const rule,
+		const struct ucred *const old_cred, const struct ucred *const new_cred)
+{
+	const int supp_ngroups = old_cred->cr_ngroups;
+	const gid_t *supp_groups = (supp_ngroups > 0) ? &new_cred->cr_groups[1] : NULL;
+
+	return (rule_grant_setgroups(rule, old_cred, supp_ngroups, supp_groups));
```

Comment:
```suggestion
	return (rule_grant_setgroups(rule, old_cred, old_cred->cr_ngroups, new_cred->cr_groups));
```

After recent changes in -CURRENT, `cr_groups` now only holds supplementary groups (and not the effective GID).